### PR TITLE
Gmp wrapper

### PR DESCRIPTION
**Pull request by fdplacido**
Added gmp-wrapper recipe

### CODE REVIEW

File: packages/gmp/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class Gmp(CMakePackage):
+    """GMP Wrapper"""
+
+    homepage = "https://github.com/fdplacido/GMP"
+    git      = "https://github.com/fdplacido/GMP.git"
+    url      = "https://github.com/fdplacido/GMP/archive/plfernan/wrapper_improvements.tar.gz"
+
+    maintainers = ['fdplacido']
+
+    version('plfernan/wrapper_improvements', branch='plfernan/wrapper_improvements')
```
Comment (vvolkl) [inline packages/gmp/package.py:17]: You could call this version`develop`, which indicates for spack that it is the newest:
https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison

File: packages/gmp/package.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+class Gmp(CMakePackage):
+    """GMP Wrapper"""
+
+    homepage = "https://github.com/fdplacido/GMP"
+    git      = "https://github.com/fdplacido/GMP.git"
+    url      = "https://github.com/fdplacido/GMP/archive/plfernan/wrapper_improvements.tar.gz"
```
Comment (vvolkl) [inline packages/gmp/package.py:13]: Point these to `https://github.com/key4hep/GMP`

File: packages/lcio/package.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ class Lcio(CMakePackage):
             description='Use the specified C++ standard when building.')
     variant("jar", default=False,
     description="Turn on to build/install lcio.jar")
-    variant("rootdict", default=True,
```
Comment (vvolkl) [inline packages/lcio/package.py:30]: Why this change? I think we agreed at one point that `+rootdict` is a good default.

File: packages/gmp/package.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    maintainers = ['fdplacido']
+
+    version('plfernan/wrapper_improvements', branch='plfernan/wrapper_improvements')
+
+    depends_on('root')
+    depends_on('lcio')
+    depends_on('marlin')
+    depends_on('gaudi')
+
+    patch('/home/plfernan/workspace/GMP/binary_tag_removal.patch')
```
Comment (vvolkl) [inline packages/gmp/package.py:24]:
```suggestion
    patch('binary_tag_removal.patch')
```

File: packages/gmp/binary_tag_removal.patch [hunk 5]
```diff
@@ -0,0 +1,51 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (vvolkl) [inline packages/gmp/binary_tag_removal.patch:1]: Did you already submit this patch as a PR to the GMP repo? Better to only apply patches to the versions where it is necessary (but since there are no released version yet, I guess it could also just be removed once it is merged).

**Comment by vvolkl** [comment]
If you follow as well the iLCSoft style of numbering v00-01 etc. you will have to add the url handler, see https://github.com/key4hep/k4-spack/blob/master/packages/gear/package.py#L42

File: packages/lcio/package.py (hunk 3 above)
Comment (fdplacido) [reply packages/lcio/package.py:30]: This one wasn't compiling for me before with that, but I just uninstalled and compiled, and its working now. I'll revert it.

**Comment by fdplacido** [comment]
> If you follow as well the iLCSoft style of numbering v00-01 etc. you will have to add the url handler, see https://github.com/key4hep/k4-spack/blob/master/packages/gear/package.py#L42

I added proper versioning for the recipe and the url_version handler.

**Comment by vvolkl** [comment]
Thanks for the updates! I think you still need to rebase your branch. I think you need still the following run environment setup because gaudi likes to install files in non-standard locations like `python/` and `scripts/`:
https://github.com/HEP-FCC/fcc-spack/blob/master/packages/fccsw/package.py#L75
Also I notice that gmprun is not installed - I'll send a patch to gmp.

For testing, I realise we need some more iLCSoft packages. we should go ahead and add some so we can actually test the installation.
